src/meshwifi/meshwifi.cpp

```diff
@@ -0,0 +1,255 @@
+#include "meshwifi.h"
+#include "NodeDB.h"
+#include "WiFiServerAPI.h"
+#include "configuration.h"
+#include "main.h"
+#include "meshwifi/meshhttp.h"
+#include <DNSServer.h>
+#include <WiFi.h>
+
+static void WiFiEvent(WiFiEvent_t event);
+// static void WiFiEvent(WiFiEvent_t event, WiFiEventInfo_t info);
+
+DNSServer dnsServer;
+static WiFiServerPort *apiPort;
+
+uint8_t wifiDisconnectReason = 0;
+
+bool isWifiAvailable()
+{
+    const char *wifiName = radioConfig.preferences.wifi_ssid;
+    const char *wifiPsw = radioConfig.preferences.wifi_password;
+
+    if (*wifiName && *wifiPsw) {
+
+        // Once every 10 seconds, try to reconnect.
+
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+// Disable WiFi
+void deinitWifi()
+{
+    /*
+        Note from Jm (jm@casler.org - Sept 16, 2020):
+
+        A bug in the ESP32 SDK was introduced in Oct 2019 that keeps the WiFi radio from
+        turning back on after it's shut off. See:
+            https://github.com/espressif/arduino-esp32/issues/3522
+
+        Until then, WiFi should only be allowed when there's no power
+        saving on the 2.4g transceiver.
+    */
+
+    WiFi.mode(WIFI_MODE_NULL);
+    DEBUG_MSG("WiFi Turned Off\n");
+    // WiFi.printDiag(Serial);
+}
+
+// Startup WiFi
+void initWifi()
+{
+    if (isWifiAvailable() == 0) {
+        return;
+    }
+
+    if (radioConfig.has_preferences) {
+        const char *wifiName = radioConfig.preferences.wifi_ssid;
+        const char *wifiPsw = radioConfig.preferences.wifi_password;
+
+        /*
+        if (0) {
+            radioConfig.preferences.wifi_ap_mode = 1;
+            strcpy(radioConfig.preferences.wifi_ssid, "MeshTest2");
+            strcpy(radioConfig.preferences.wifi_password, "12345678");
+        } else {
+            radioConfig.preferences.wifi_ap_mode = 0;
+            strcpy(radioConfig.preferences.wifi_ssid, "meshtastic");
+            strcpy(radioConfig.preferences.wifi_password, "meshtastic!");
+        }
+         */
+
+        if (*wifiName && *wifiPsw) {
```

Comment:
I think this is fine for now, but I think wifiPsw should be a llowed to be blank - for connecting to open APs.

Reply:
Oh yeah, that's just for my local work. I don't have an easy way to change those values since I need to drop to Linux to use the Python API.

That block of code is actually commented out. The values for those will be taken from whatever has been saved to the device with the API.

I'll take it out in my next check-in.